addons/mass_mailing/models/mailing.py

```diff
@@ -470,47 +470,24 @@ def _get_seen_list(self):
         self.ensure_one()
         target = self.env[self.mailing_model_real]
 
```

Comment:
Note: now, unless I'm missing something, this looks like it's basically equivalent to
```
domain = [('email', '!=', False), ('model', '=', self.mailing_model_real)]
if self.unique_ab_testing:
     domain = AND([domain, [('campaign_id', '=', self.campaign_id)]])
else:
     domain = AND([domain, [('mass_mailing_id', '=', self.id)])
return self.env['mailing.trace'].sudo().search(domain).mapped('email')
``` 
with probably no huge performance benefit.

Might be good to rework that, probably not here though. You can leave it as is so we can forward the fix without worrying about it, just seemed interesting.

Reply:
I agree, might be interesting to do in master as to not change stable versions too much